public/app/plugins/datasource/loki/components/monaco-query-field/MonacoQueryField.tsx

```diff
@@ -125,8 +130,13 @@ const MonacoQueryField = ({ languageProvider, history, onBlur, onRunQuery, initi
             if (!model) {
               return;
             }
-
-            const errors = validateQuery(model.getValue(), model.getLinesContent()) || [];
+            const query = model.getValue();
+            const errors =
+              validateQuery(
+                query,
+                datasource.interpolateString(query, placeHolderScopedVars),
+                model.getLinesContent()
+              ) || [];
```

Comment:
The issue with this approach is that if we interpolate values and they have different length than their variable, the error highlight is at wrong position 
<img width="703" alt="image" src="https://user-images.githubusercontent.com/30407135/205326160-90ee7fd7-aded-4e76-a310-9335d9e079a9.png">
<img width="836" alt="image" src="https://user-images.githubusercontent.com/30407135/205326352-dfb0e070-56bc-4a57-8e24-7673e51a25c2.png">

So we should probably adjust the position, if you want to validate also variables.

Reply:
But I personally think, that we could just "ignore" template variables and maybe decide in the future if we want to enhance validation to include interpolating of variables.

Reply:
Thanks for the feedback. I have a few more ideas to explore, so I'll give this another try. As a last resource, we can ignore variables and consider variables as a future enhancement.

Reply:
@ivanahuckova I changed my approach a little bit, and got more accurate results when validating interpolated queries.

I still think we should support it, to provide the best experience validating queries (i.e. minimizing the amount of failed queries reaching the backend), so I look forward to keep iterating here.